### ISSUE TITLE
Non-text files not displayed in folder manager

### ISSUE THREAD

**Issue by davidmhewitt** [elementary/code#449]
Files such as images or fonts aren't displayed in the folder tree, which I find slightly confusing. I realise this is intentional to stop you trying to open binary files in a text editor, but there are a couple of context menu items like "delete" and "open in" that would still be useful and prevent me having to open my file manager instead.

I guess there just needs to be a design decision as to whether or not to display them and how exactly they should be handled.

**Comment by tintou** [comment]
The side-view doesn't replace the File Explorer application, this has been handled this way by design as it's already covered by the current code.

**Comment by danrabbit** [comment]
I think I'd like to reopen this for discussion. Something to think about for example is if we have Git integration in the future, it would be useful to be able to see non-code files for that purpose

**Comment by cassidyjames** [comment]
Implementation-wise, I think if they're not an editable text file, we should just open them in the default handler when they're clicked on.

**Comment by jlnr** [comment]
The current behavior looks like a bug when expanding folders that only contain non-code files: there's an empty item that does nothing when selected.

![code-empty-folders](https://user-images.githubusercontent.com/70772/46479006-64acf300-c7ee-11e8-920a-98859505093e.png)

I would prefer to be able to see and open non-code files right from the sidebar, as suggested above.

**Comment by cassidyjames** [comment]
In the case where editable (text) and non-editable (binary) files are side-by-side in a folder, would we want to denote that or separate them somehow? It could be confusing as to why some files open in Code and others don't. Or perhaps the mimetype icon is enough info here? Thoughts from @elementary/ux?

**Comment by cassidyjames** [comment]
I do agree that the current behavior looks like a bug in some contexts, and with git tracking I'd want all files displayed there so I can see and track their status.

**Comment by wout** [comment]
Even better would be if, in case of images, a preview would be given. Just like Atom handles image files:

![image](https://user-images.githubusercontent.com/107324/48426417-09fca480-e767-11e8-9d84-2c409af3309e.png)

If at all possible of course.

**Comment by cassidyjames** [comment]
@wout I'm strongly opposed to that, because reimplementing a file previewer in the code editor not only bloats the code editor, but it goes against the overall philosophy of elementary OS where dedicated apps handle their respective content types. Opening the Photos app to preview something is just as fast as opening a reimplemented preview view in the editor, and reduces the need for yet another click or keypress to get it open into the app you'd expect.

**Comment by cassidyjames** [comment]
@micahilbery I'd be concerned that insensitive looks strange, but that might work. I'd be interested in seeing it I suppose.

**Comment by wout** [comment]
@cassidyjames I agree with you. In hindsight, there is no added value in previewing image images in a code editor. Thanks for your feedback.

**Comment by roymckenzie** [comment]
@cassidyjames here is a sample of what it could look like:

![All file types shown](https://user-images.githubusercontent.com/1065321/54100181-cd426200-4382-11e9-807f-18290899b906.png)

The arrow-character badge isn't that great. Maybe we could just get rid of it and when a user clicks the item it reveals the context menu?

**Comment by cassidyjames** [comment]
@roymckenzie I don't think we need an added visual indicator there… we already have the icon which shows the filetype, and non-text filetypes should just be opened in the system default app for that filetype.

**Comment by roymckenzie** [comment]
@cassidyjames Ok great. I'll fix that up with a PR!

**Comment by cassidyjames** [comment]
Thanks for the work, @roymckenzie!
